### ISSUE TITLE
`t2 update` with name flag will cause update to fail

### ISSUE THREAD

**Issue by johnnyman727** [tessel/t2-cli#608]
Reported on the forums:
```
LC-DEV-MACBOOKPRO:server jdudley$ t2 update -version 0.0.11 --name RAS2
ERR! Error: SSH key path must be a string type.
LC-DEV-MACBOOKPRO:server jdudley$ t2 update --name RAS2
INFO Looking for your Tessel...
INFO Connected to RAS2.
WARN Must have Tessel connected over USB to complete update. Aborting update.
LC-DEV-MACBOOKPRO:server jdudley$ t2 update --name RAS2
INFO Looking for your Tessel...
INFO Connected to RAS2.
WARN No LAN connection found. USB-only updates do not work yet. Please ensure Tessel is connected to wifi and try again
```

This doesn't seem to happen when the name flag is not included.

**Comment by johnnyman727** [comment]
The issue may lie with the `--version` option as well

**Comment by johnnyman727** [comment]
So there are actually two issues here. In the submitted example, you can see that the `--version` flag was only passed in with one `-` (as `-version`) so it wasn't parsed properly. `-i` is the option for provided a SSH keyPath but because the command isn't formatted properly with a space after the `i`, it just treats `key` as a flag which is true. I'm not sure there is any way to get around this. We could check for the option being `true` or `false` rather than a path, I suppose.

The real issue is the `--name` flag. When the `--name` flag is passed in, the CLI stops searching for Tessels as soon as it finds one with the proper name. USB devices will almost always be found first so it resolves as soon as it has a USB device and the LAN connection is never found. Unfortunately, `update` requires a USB **and** LAN connection right now for updates to work so providing a `--name` flag will ensure that `update` fails.

**Comment by johnnyman727** [comment]
Fixed as of `0.0.18` where you can use a USB connection to update.
